apis/core/bootstrap/shapes/dimension.ts

```diff
@@ -83,18 +123,92 @@ ${shape('dimension/metadata')} {
       ${sh.minLength} 1 ;
       ${dash.singleLine} false ;
       ${sh.order} 30 ;
-    ], [
+    ] , [
+      ${sh.name} "Data kind" ;
+      ${sh.path} ${cube.dataKind} ;
+      ${sh.maxCount} 1 ;
+      ${sh.nodeKind} ${sh.BlankNode} ;
+      ${sh.node} [
+        ${sh.property} [
+          ${sh.name} "Choose type" ;
+          ${sh.path} ${rdf.type} ;
+          ${sh.minCount} 1 ;
+          ${sh.maxCount} 1 ;
+          ${sh.nodeKind} ${sh.IRI} ;
+          ${dash.editor} ${dash.EnumSelectEditor} ;
+          ${sh.in} (
+            ${schema.GeoCoordinates}
+            ${schema.GeoShape}
+            ${time.GeneralDateTimeDescription}
+          ) ;
+          ${sh.order} 10 ;
+        ] , [
+          ${sh.name} "Time precision" ;
+          ${sh.path} ${time.unitType} ;
+          ${sh.maxCount} 1 ;
+          ${sh.nodeKind} ${sh.IRI} ;
+          ${sh1.if} [
+            ${sh.path} ${rdf.type} ;
+            ${sh.hasValue} ${time.GeneralDateTimeDescription} ;
+          ] ;
```

Comment:
Custom annotation to allow `time.unitType` only when `?focusNode rdf:type time:GeneralDateTimeDescription`

Reply:
Not a fan of the prefix, but the feature looks great! I think it will allow me to do some stuff that I avoided until now :)

Reply:
Just curious, could http://datashapes.org/constraints.html#HasValueWithClassConstraintComponent be used for a similar purpose?

Reply:
Did not know that. Close but I think it's a little different